### DISCUSSION TITLE
Multiple Image Upload

### DISCUSSION THREAD

**Discussion by photojo** [filamentphp/filament#6285] [Help] [answered]
I use 

`

protected function getTableActions(): array
    {
        return [
            EditAction::make('edit')    
                ->action(function (Product $record, array $data) {
                    $record->image = $data['image'];
                    $record->save();
                })
                ->form([
                        FileUpload::make('image')
                        ->disk('local')
                        ->preserveFilenames()
                        ->multiple()
                        ->directory('myfiles')
                        ->visibility('private')
                        ->storeFileNamesIn('attachment_file_names')
                        ->acceptedFileTypes(['image/jpeg'])
                        ->enableOpen(), ...

`

to upload multiple image-files. It works fine and I can display images f.e. in blade-files.

In DB I get: ["myfiles\/img_car.JPG","myfiles\/img_bike.JPG"]

**But**, if I edit the same row again, I can not see the images? I always have to upload all images again?!

**Comment by danharrin** [comment]
Yup the images that are saved to the database will be filled into the form again. Must be something wrong on your end. Can you send a screenshot of what happens when you load the images in? Any console errors in the browser?

**Reply by photojo** [reply]
Sorry, for my delayed answer. I did some more tests. 
I can not see any errors in console or in debug-bar.
<img width="808" alt="screenshot_upload" src="https://user-images.githubusercontent.com/28451713/233769936-9c5e4388-f7fd-40f0-975f-b6508fc292d7.png">

**Comment by photojo** [comment]
Maybe it's due to that images are not stored in public-folder ?!

**Reply by danharrin** [reply]
They get stored when you submit the form

**Reply by photojo** [reply]
Ok, but why can I not see the files in 'modl', when edit data-record again?

**Reply by danharrin** [reply]
Do they get saved ok to the database? Do you have an `array` cast on that field in the model?

**Comment by photojo** [comment]
Yes, they are saved in database.

And in my model I have  
protected $casts = ['image'=> 'array'];

**Reply by danharrin** [reply]
I don't know then. Check out the demo, they save fine

**Reply by photojo** [reply, marked as answer]
Ok, I solved it. Made all new, cause I did not found the error!
Now, it works fine!
Thank you very much, Dan!

**Comment by vitalijalbu** [comment]
Hi guys, I have 2 tables:
1. listings
2. photos

I want to use original FileUpload component, meaning I will store my images into photos table.
Images upload correctly , but how I can save the data inside Photo.php model after upload?

`                FileUpload::make('photos')
                    ->disk('public')
                    ->columnSpan('full')
                    ->multiple()
                    ->minFiles(1)
                    ->maxFiles(15)
                    ->reorderable()
                    ->visibility('public'),`


`class Photo extends Model
{
    use HasFactory;

    protected $fillable = [
        'listing_id',
        'uid',
        'name',
        'file_name',
        'position'
];
}`

**Reply by jaskaran-ai** [reply]
Looking for same

**Reply by devope** [reply]
This is what I did:

```php
->afterStateUpdated(
    function (callable $get, callable $set) {

        $property = Property::find($get('id'));
        $images = $get('images');

        // Remove old images
        $property->images()->delete();

        // Add new images
        foreach ($images as $image) {
            $property->images()->create([
                'path' => $image->getFilename()
            ]);
        }
    }
)
```

But I can't understand how display them on edit.

**Comment by photojo** [comment]
I do not understand your problem exactly. Could you describe it again?

**Comment by kasuya100** [comment]
I am using Laravel 10 and FilamentPHP 3.2
I have a problem with grid 
![Captura de pantalla 2025-03-21 034538](https://github.com/user-attachments/assets/d3ed603f-14b5-4070-a0a9-b58bb0af477b)
but list looks good
![Captura de pantalla 2025-03-21 034655](https://github.com/user-attachments/assets/28e0e8a8-d3a3-4218-85bc-5790044a872c)
